### ISSUE TITLE
Mobile allows CRs in title which is inconsistent behaviour with desktop + can lead to data loss

### ISSUE THREAD

**Issue by Daeraxa** [laurent22/joplin#5482]
### Mobile
Joplin 2.3.4
Database v39
Android 11
**USING BETA EDITOR**
### Desktop
Joplin 2.3.5 (prod, win32)
Client ID: a4ed20b9be424a1c8c3e5f3cc355c345
Sync Version: 3
Profile Version: 39
Keychain Supported: Yes
Revision: 819af3c
Windows 10

## Steps to reproduce

1.  Create new note and start typing title manually
2. Press 'return' and continue typing - repeat
3. Move cursor to note body and type something
4. Return from note and sync
5. Access new note on desktop app - Title display is different from mobile and data has been lost
6. Edit note on desktop + sync
7. Sync mobile + access note - Now only has single title line, overflow has been moved to body but the "1CR" line has been lost

### Mobile seems happy to display the title including the CRLF:
Notes panel:
<img src="https://user-images.githubusercontent.com/58074586/133943512-fcb7fa48-297f-465a-aa6b-b4de1c003052.png" width="500">
Viewer:
<img src="https://user-images.githubusercontent.com/58074586/133943621-3fe593f9-621a-404c-8ee0-f1b3f479f466.png" width="500">

### Desktop puts extra lines into the body of the note (but not all of it - "1CR" is missing):
![image](https://user-images.githubusercontent.com/58074586/133943526-5ad46bc6-cde2-48e9-8456-66b87dd6df5a.png)

### Content of desktop database ("1CR" has been lost):
```sql
sqlite> SELECT title,
   ...>        body
   ...>   FROM notes
   ...>  WHERE id = "db7daac4e933415c93aaf1c0939df9b1";
```
```
Title|2cr

Line after return

Line after return and 2 spaces


Note body line 1
```

### Content of sync target md file:
```
Title 1cr 2cr

Line after return

Line after return and 2 spaces

Note body line 1
```

## Describe what you expected to happen

Mobile should not be allowed to put CRs into titles. If it is allowed to then all the data should be visible to the desktop app - current behaviour casuses some data to be lost.
Behaviour of pressing return on mobile keyboard should move cursor to note body

## Logfile

[syncReport-1632085837890.txt](https://github.com/laurent22/joplin/files/7192954/syncReport-1632085837890.txt)
[log.txt](https://github.com/laurent22/joplin/files/7192953/log.txt)

**Comment by Daeraxa** [comment]
For sake of completion I created a new note with the same content and checked the mobile db which shows the full content within notes.title.
```sql
sqlite> select title, body from notes where id="f01f47bdf8a64ac59878f18018d6e73f";
Title2
1cr
2cr

Line after return

Line after return and 2 spaces
|Note body line 1
```

**Comment by Daeraxa** [comment]
I think I might have been able to solve this myself but I want to check a few things before I submit a PR as it appears the multiline functionality in the title was intentional - but only for Android...

So the "fix" was simply to change this line:
https://github.com/laurent22/joplin/blob/3ddfa0b774fc40da6b2bb4404f58d6c1c4363b79/packages/app-mobile/components/screens/Note.tsx#L1182
to `{false}`

However it is clearly set up intentionally for platforms that aren't ios?

https://github.com/laurent22/joplin/blob/3ddfa0b774fc40da6b2bb4404f58d6c1c4363b79/packages/app-mobile/components/screens/Note.tsx#L96-L97

https://github.com/laurent22/joplin/blob/3ddfa0b774fc40da6b2bb4404f58d6c1c4363b79/packages/app-mobile/components/screens/Note.tsx#L392

https://github.com/laurent22/joplin/blob/3ddfa0b774fc40da6b2bb4404f58d6c1c4363b79/packages/app-mobile/components/screens/Note.tsx#L974-L975

So basically, I'm happy to submit a simple PR to change that single line (it does stop you making multiple lines when tested on an android emulator (side note - that was fun to set up........)) but the fact it seems to have been implemented intentionally is what is throwing me. Is anyone aware of why this exists and was implemented as Android only functionality?

**Comment by pathange-s** [comment]
@laurent22 Is this issue still open and up for snatch ? I see that there was some discussion on this being implemented intentionally. May I know the status ?

**Comment by Daeraxa** [comment]
If the answer is provided I'd rather finish this and close it myself.

**Comment by laurent22** [comment]
@Daeraxa, you're right multiline titles used to be supported although that's probably not a very good idea. So yes feel free to remove that code and create a pull request. Actually please could you also remove that enableMultilineTitle_ variable any related code (i.e. assuming it's always false now, some code can be removed because it will never be executed).
